src/providers/asciidoc.provider.ts

```diff
@@ -0,0 +1,97 @@
+import * as vscode from "vscode";
+import * as path from "path";
+import { createContext, Context } from "./createContext";
+
+import { createPathCompletionItem } from "./createCompletionItem";
+import {
+  getPathOfFolderToLookupFiles,
+  getChildrenOfPath,
+  sortFilesAndDirectories,
+} from "../util/file";
+
+export const AsciidocProvider = {
+  provideCompletionItems,
+};
+
+export async function provideCompletionItems(
+  document: vscode.TextDocument,
+  position: vscode.Position
+): Promise<vscode.CompletionItem[]> {
+  const context = createContext(document, position);
+
+  return shouldProvide(context)
+    ? provide(context)
+    : Promise.resolve([]);
+}
+
+/**
+ * Checks if we should provide any CompletionItems
+ * @param context
+ */
+function shouldProvide(context: Context): boolean {
+  return /(include\:\:|image\:\:|image\:)\S*/gi.test(context.textFullLine)
+}
+
+/**
+ * Provide Completion Items
+ */
+async function provide(
+  context: Context
+): Promise<vscode.CompletionItem[]> {
+  const documentText = context.document.getText();
+  const pathExtractedFromIncludeString = context.textFullLine.replace('include::', '').replace('image::', '').replace('image:', '');
+  let entryDir = pathExtractedFromIncludeString.substr(0, pathExtractedFromIncludeString.lastIndexOf("/"));
+
+  // use path defined in a variable used
+  if (entryDir.startsWith('{')) {
+    const variableName = entryDir.replace('{', '').replace('}', '')
+    const match = documentText.match(new RegExp(`:${variableName}:.*`, 'g'));
+    if (match && match[0]) {
+      entryDir = match[0].replace(`:${variableName}: `, '');
+    }
+  }
+
+  const workspace = vscode.workspace.getWorkspaceFolder(context.document.uri);
+  const rootPath = workspace?.uri.fsPath;
+  const searchPath = getPathOfFolderToLookupFiles(
+    context.document.uri.fsPath,
+    path.join(rootPath, entryDir)
+  );
+
+  const childrenOfPath = await getChildrenOfPath(searchPath);
+
+  const items = sortFilesAndDirectories(childrenOfPath);
+
+  const levelUpCompletionItem: vscode.CompletionItem = {
+    label: '..',
+    kind: vscode.CompletionItemKind.Folder,
+    sortText: '..',
+  }
+  const globalVariableDefinitions = documentText.match(/:\S+:.*/g)
+
+  let variablePathSubstitutions = []
+  // TODO: prevent editor.autoClosingBrackets at this point until finished inserting
+  const editorConfig = vscode.workspace.getConfiguration('editor');
+  const doAutoCloseBrackets = editorConfig.get('autoClosingBrackets') === 'always';
+  if (globalVariableDefinitions) {
```

Comment:
We already have autocompletion on the brace character `{` from the Asciidoc AST (see https://github.com/asciidoctor/asciidoctor-vscode/blob/master/src/features/attributeCompleter.ts).